ext/standard/string.c

```diff
@@ -2032,7 +2009,7 @@ PHP_FUNCTION(strripos)
 		Z_PARAM_LONG(offset)
 	ZEND_PARSE_PARAMETERS_END_EX(RETURN_FALSE);
 
-	if (ZSTR_LEN(haystack) == 0 || ZSTR_LEN(needle) == 0) {
+	if (ZSTR_LEN(haystack) == 0 || ZSTR_LEN(needle) > ZSTR_LEN(haystack)) {
```

Comment:
This will give a wrong result for `strrpos("", "")` I believe.

Reply:
Commented out the condition and tests still pass green.
Will remove it and commit it.